pandas/io/parsers/arrow_parser_wrapper.py

```diff
@@ -150,6 +151,16 @@ def read(self) -> DataFrame:
             parse_options=pyarrow_csv.ParseOptions(**self.parse_options),
             convert_options=pyarrow_csv.ConvertOptions(**self.convert_options),
         )
-
-        frame = table.to_pandas()
```

Comment:
Do you know if the ``types_mapper`` kwarg would work here in the to_pandas call?

Reply:
`types_mapper` works, but it requires a conversion from arrow -> pandas -> back to arrow.

The implementation here just directly converts arrow table -> arrow ChunkedArray and sticks it into pandas